### DISCUSSION TITLE
How to evaluate mdx with react shortcodes in browser

### DISCUSSION THREAD

**Discussion by fr-an-k** [mdx-js/mdx#2058] [Q&A] [answered]
You can use MDXProvider with shortcodes, but that seems to be server-side.
How would I go about doing the same in the browser, using the evaluate function from @mdx-js/mdx?
Would I need to provide a unified plugin or something?

**Comment by wooorm** [comment, marked as answer]
Shortcodes (“components”) is not server side. `evaluate` can indeed be used. You don’t need a plugin.

**Reply by fr-an-k** [reply]
Thanks, but how can I use evaluate to provide shortcodes linked to react components?

**Reply by wooorm** [reply]
I do not know what you mean. Please spend time framing your question as detailed in the [support guide](https://mdxjs.com/community/support/).

You should be able to understand what you need to do by reading the [Using MDX](https://mdxjs.com/docs/using-mdx/) guide and the API reference of [`@mdx-js/mdx`](https://mdxjs.com/packages/mdx/). Without knowing your problem, my guess is that the answer is: pass the components to the `MDXContent` component you get back.

**Reply by fr-an-k** [reply]
I got it now, great thanks!
